### PR TITLE
Feature: helm chart to deploy cluster quickly

### PR DESCRIPTION
**Pull request by PowerfooI**
## Summary

Helm chart to deploy oceanbase cluster which auto generate required user secrets.

### CODE REVIEW

File: charts/oceanbase-cluster/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+#
+# Library charts provide useful utilities or functions for the chart developer. They're included as
+# a dependency of application charts to inject those utilities and functions into the rendering
+# pipeline. Library charts do not define any templates and therefore cannot be deployed.
+type: application
+
+# This is the chart version. This version number should be incremented each time you make changes
+# to the chart and its templates, including the app version.
+# Versions are expected to follow Semantic Versioning (https://semver.org/)
+version: 4.2.0
```
Comment (chris-sun-star) [inline charts/oceanbase-cluster/Chart.yaml:18]: maybe this version is too short, keep this version the same as observer version

File: charts/oceanbase-cluster/templates/obcluster.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    resource:
+      {{- toYaml .resource | nindent 6 }}
+    storage:
+     {{- range $key, $size := .storages }} 
+      {{ $key }}:
+        storageClass: {{ $.Values.storageClass }}
+        size: {{ $size }}
+     {{- end }}
+    {{- end }}
+  monitor:
```
Comment (chris-sun-star) [inline charts/oceanbase-cluster/templates/obcluster.yaml:29]: monitor is also optional, add a parameter to specific whether to enable monitor but default true

File: charts/oceanbase-cluster/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+apiVersion: v2
```
[the remark this answers is not in the document]
Comment (PowerfooI) [reply charts/oceanbase-cluster/Chart.yaml:1]: This api version is used by Helm. In Helm3, the field should be `v2`, it specify the helm version.

File: charts/oceanbase-cluster/Chart.yaml (hunk 1 above)
Comment (PowerfooI) [reply charts/oceanbase-cluster/Chart.yaml:18]: This field should follow **semver** standard, but our 4.2.0.0-YYYYMMDDHHMMss format is invalid in such style.

File: charts/oceanbase-cluster/Chart.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (PowerfooI) [reply charts/oceanbase-cluster/Chart.yaml:1]: image

File: charts/oceanbase-cluster/templates/obcluster.yaml (hunk 2 above)
Comment (PowerfooI) [reply charts/oceanbase-cluster/templates/obcluster.yaml:29]: OK, I'll do that ~

File: charts/oceanbase-cluster/Chart.yaml (hunk 1 above)
Comment (PowerfooI) [reply charts/oceanbase-cluster/Chart.yaml:18]: 4.2.0-YYYYMMDDHHMMss is correct, but I think the forth number can not be ignored.

Comment (PowerfooI) [reply charts/oceanbase-cluster/Chart.yaml:18]: or `4.2.0-subpatch.0-101000032023091319`. Major version pattern in semver is "X.Y.Z" which contains three parts, the forth number should be transformed to another form.

File: charts/oceanbase-cluster/Chart.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (chris-sun-star) [reply charts/oceanbase-cluster/Chart.yaml:1]: ok, then oboperator chart also needs update api to v2

**Review by chris-sun-star — APPROVED** [review]

**Pull request merged**